platforms/software/testing-base/src/integTest/groovy/org/gradle/testing/CustomTestTaskHTMLTestReportMetadataTest.groovy

```diff
@@ -339,4 +345,9 @@ final class CustomTestTaskHTMLTestReportMetadataTest extends AbstractIntegration
             tasks.register("${name}", ${name}CustomTestTask)
         """
     }
+
+    private String normalizePath(String relativePath) {
+        boolean isWindows = System.getProperty("os.name").toLowerCase().contains("windows")
+        return (isWindows ? "/" : "") + TextUtil.normaliseFileSeparators(testDirectory.file(relativePath).absolutePath.replace(" ", "%20"))
```

Comment:
I think this can just be `file(relativePath).toURI().toASCIIString()`

Reply:
This doesn't work with our non-ASCII test directory: `tešt files`.

Reply:
OK, maybe just `toString` works. I was thinking that the href's had to be ASCII only